src/System.Text.Encoding.CodePages/tests/EncodingCodePages.cs

```diff
@@ -2,911 +2,581 @@
+                new int[] { 0x41, 0x42, 0x43, 0x3002, 0x6F86, 0x44, 0x45 }
+            };
 
-        bytes = new byte[] { 0x5B, 0x5C, 0x5D, 0x5E, 0x7B, 0x7C, 0x7D, 0x7E };
-        unicodeStr = "\x005B\x005C\x005D\x005E\x007B\x007C\x007D\x007E";
-        TestRoundtrippingCodepageEncoding("us-ascii", bytes, unicodeStr);
-    }
+            yield return new object[] { "iso-2022-kr",
+                new byte[] { 0x0E, 0x21, 0x7E, 0x1B, 0x24, 0x29, 0x43, 0x21, 0x7E, 0x0F, 0x21, 0x7E, 0x1B, 0x24, 0x29, 0x43, 0x21, 0x7E },
+                new int[] { 0xFFE2, 0xFFE2, 0x21, 0x7E, 0x21, 0x7E }
+            };
 
-    [Fact]
-    public static void TestCodepageEncoding()
-    {
-        EnsureInitialization();
+            yield return new object[] { "hz-gb-2312",
+                new byte[] { 0x7E, 0x42, 0x7E, 0x7E, 0x7E, 0x7B, 0x21, 0x7E, 0x7E, 0x7D, 0x42, 0x42, 0x7E, 0xA, 0x43, 0x43 },
+                new int[] { 0x7E, 0x42, 0x7E, 0x3013, 0x42, 0x42, 0x43, 0x43, }
+            };
+        }
 
-        foreach (var id in s_encodingIdTable)
+        private static IEnumerable<KeyValuePair<int, string>> CrossplatformDefaultEncodings()
         {
-            // Console.WriteLine(id.Name);
-            Encoding enc = Encoding.GetEncoding(id.Name);
-            byte[] bytes = enc.GetBytes("Az");
-            char[] chars = enc.GetChars(bytes);
-            Assert.True(chars.Length == 2 && chars[0] == 'A' && chars[1] == 'z');
-            // Encoding.GetEncoding(id.Codepage);
+            yield return Map(1200, "utf-16");
+            yield return Map(12000, "utf-32");
+            yield return Map(20127, "us-ascii");
+            yield return Map(65000, "utf-7");
+            yield return Map(65001, "utf-8");
         }
-    }
-
-    [Fact]
-    public static void TestDecodingIso2022()
-    {
-        EnsureInitialization();
-
-        byte[] japaneseEncodedBuffer = new byte[]
-            {
-                0xA,
-                0x1B, 0x24, 0x42, 0x25, 0x4A, 0x25, 0x4A,
-                0x1B, 0x28, 0x42,
-                0x1B, 0x24, 0x42, 0x25, 0x4A,
-                0x1B, 0x28, 0x42,
-                0x1B, 0x24, 0x42, 0x25, 0x4A,
-                0x1B, 0x28, 0x42,
-                0x1B, 0x1, 0x2, 0x3, 0x4,
-                0x1B, 0x24, 0x42, 0x25, 0x4A, 0x0E, 0x25, 0x4A,
-                0x1B, 0x28, 0x42, 0x41, 0x42, 0x0E, 0x25, 0x0F, 0x43
-            };
-
-        string codepageName = "iso-2022-jp";
-        Decoder jpDecoder = Encoding.GetEncoding(codepageName).GetDecoder();
-        const int BUFFER_SIZE = 100;
-        char[] buffer = new char[BUFFER_SIZE];
 
-        int byteIndex = 0;
-        int charIndex = 0;
+        private static KeyValuePair<int, string> Map(int codePage, string webName)
+        {
+            return new KeyValuePair<int, string>(codePage, webName);
+        }
 
-        while (byteIndex < japaneseEncodedBuffer.Length)
+        [Fact]
+        public static void TestDefaultEncodings()
```

Comment:
Also, FYI, when I pull down your changes and run them locally on my Linux VM, I get an additional failure:
```
Test.EncodingTest.TestCodepageEncoding(codePage: 29001, webName: "x-europa", queryString: "x-europa") [FAIL]
      System.NotSupportedException : No data is available for encoding 29001.
      Stack Trace:
            at System.Text.EncodingNLS.get_WebName()
            at Test.EncodingTest.TestCodepageEncoding(Int32 codePage, String webName, String queryString)
```
Not sure why that didn't show up in the initial CI run, but maybe it will in a subsequent.